creator/ingest_runs/tasks/validation_run.py

```diff
@@ -28,51 +32,108 @@
 logger = logging.getLogger(__name__)
 
 
-class ExtractConfigFileDoesNotExist(Exception):
+class ExtractDataError(Exception):
+    """
+    Raise when something goes wrong in creator.analyses.analyzer.extract_data
+    """
     pass
 
 
-def clean_and_map(version: Version) -> pandas.DataFrame:
+def version_display(version):
     """
-    Load tabular data from file version into a DataFrame and then use
-    ingest lib's extraction utility to clean the values and map the column
-    names to the standard set of concepts expected by the validator
+    Helper to display version id and file name for log statements
     """
-    logger.info(f"Performing extraction to clean and map file {version}")
+    return f"{version.pk}: {version.file_name}"
 
-    # Load ingest extract operations for this file
-    if not version.root_file:
-        raise Exception(
-            f"Validation failed. Version {version} missing root_file"
-        )
-    ec = version.extract_config_path
-    if not ec:
+
+def generate_mapper(
+    template_versions: List[TemplateVersion]
+) -> Dict[str, str]:
+    """
+    Generate a mapping dict that maps columns in the study templates to
+    template keys. This will be used when mapping source file columns to
+    template keys in preparation for file validation
+    """
+    # Create a map of template columns to template keys
+    # Also add a mapping of template keys to template keys in case the source
+    # file has already been mapped to the template keys
+    columns_to_keys = {}
+    for tv in template_versions:
+        for field in tv.field_definitions["fields"]:
+            key = field.get("key")
+            if key:
+                columns_to_keys[field["label"]] = key
+                columns_to_keys[key] = key
+
+    # Check for duplicate mappings (multiple source columns map to the same
+    # template key) and log them since this shouldn't really happen but if
+    # it does, could mean inconsistent data in the source file
+    reverse_mapping = defaultdict(set)
+    for src, key in columns_to_keys.items():
+        reverse_mapping[key].add(src)
+    duplicates = {
+        key: src_cols
+        for key, src_cols in reverse_mapping.items()
+        if len(src_cols) > 2
+    }
+    if duplicates:
```

Comment:
The above logic may be more clearly expressed as:
```
duplicates = Counter(columns_to_keys.values())
if max(duplicates.values()) > 2:
  logger.warning()
```

Reply:
why is it `len(src_cols) > 2`? should it be `len(src_cols) > 1`?

if we want to keep the log message the same, could do

```
vals = set()
duplicates = {}
for key, val in columns_to_keys.items():
    if val not in vals:
        vals.add(val)
    else:
        duplicates[key] = val
if duplicates:
    logger.warning(
        "Found multiple columns in source file(s) that map to the same "
        f"template key:\n{pformat(duplicates)}"
    )
```

Reply:
why is it `len(src_cols) > 2`?

Reply:
I like this approach for the case where I don't care about displaying src cols with the same mapped template key

Reply:
> why is it len(src_cols) > 2

@gsantia  Bc every template column gets mapped to a template key and then I also added a mapping from every template key to template key. I did this bc I want to be able to extract columns from the src file which match template columns or template keys